### PR TITLE
Show a map on AddDestination.tsx

### PR DESCRIPTION
**Pull request by derjubu**
Part of #16 

- Input field ist shown on AddDestination.tsx
- Map is shown
- On entering a destination, a marker is set on map and according part of the map is shown
- Hitting the 'Go' button leads to the next page and puts the data into local storage

Please be not be deterred by the +999 lines of code; 850 are from the new packages that were installed 😬

### CODE REVIEW

File: src/app/components/LocationMap/MapwithMarker.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    accessToken: mapboxgl.accessToken,
+    marker: false,
+  });
+
+  geocoder.on('result', (result) => {
+    marker.remove;
+    const markerPoint = result.result.center;
+    map.current ? marker.setLngLat(markerPoint).addTo(map.current) : null;
+    map.current?.flyTo({ center: markerPoint });
+  });
```
Comment (lmachens) [inline src/app/components/LocationMap/MapwithMarker.tsx]: You create an event handler on each render. 
Instead, I would create a single `geocoder` on mount (`useEffect`).

File: src/app/components/LocationMap/MapwithMarker.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  const geocoder = new MapboxGeocoder({
+    accessToken: mapboxgl.accessToken,
+    marker: false,
+  });
+
+  geocoder.on('result', (result) => {
+    marker.remove;
+    const markerPoint = result.result.center;
+    map.current ? marker.setLngLat(markerPoint).addTo(map.current) : null;
```
Comment (lmachens) [inline src/app/components/LocationMap/MapwithMarker.tsx]: See https://javascript.info/ifelse#non-traditional-use-of

File: src/app/components/LocationMap/MapwithMarker.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  const [zoom] = useState(9);
+  const marker = new mapboxgl.Marker();
+
+  const geocoder = new MapboxGeocoder({
+    accessToken: mapboxgl.accessToken,
+    marker: false,
+  });
+
+  geocoder.on('result', (result) => {
+    marker.remove;
```
Comment (lmachens) [inline src/app/components/LocationMap/MapwithMarker.tsx]: Doesn't do anything :)

```suggestion
```

File: src/app/components/LocationMap/MapwithMarker.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+};
+
+export default function MapWithMarker({
+  displayArea,
+}: MapWithMarkerProps): JSX.Element {
+  if (typeof import.meta.env.VITE_MAPBOX_ACCESS_KEY === 'string') {
+    mapboxgl.accessToken = import.meta.env.VITE_MAPBOX_ACCESS_KEY;
+  } else {
+    throw new Error('no KEY provided');
+  }
```
Comment (lmachens) [inline src/app/components/LocationMap/MapwithMarker.tsx]: It's uncommon to check if environment variables are set during render cycle.
Instead, you could move this logic to outside `MapWithMarker`.

File: src/app/pages/AddDestination/AddDestination.tsx [hunk 5]
```diff
@@ -36,38 +37,42 @@ export default function AddDestination(): JSX.Element {
...
+          <InputField
+            type="date"
+            id="end-trip"
+            value={endDate}
+            onChange={(event) => setEndDate(event.target.value)}
+          />
+        </InputLabel>
+        <DefaultButton>Go</DefaultButton>
+      </DestinationForm>
+      <MapWithMarker displayArea={'#MAP'} />
```
Comment (lmachens) [inline src/app/pages/AddDestination/AddDestination.tsx]: If you want to use `newLocation` in your `MapWithMarker`, you need to forward it by adding a property.
Maybe `<MapWithMarker displayArea={'#MAP'} location={newLocation} />`

**Review by lmachens — COMMENTED** [review]
Good job

File: src/app/components/LocationMap/LocationMap.tsx [hunk 6]
```diff
@@ -52,7 +52,7 @@ const MapContainer = styled.div`
 `;
 
 const MapLegend = styled.div`
-  position: absolute;
+  position: relative;
```
Comment (PReichetanz) [inline src/app/components/LocationMap/LocationMap.tsx:55]: Does too and left do something if the position is now relative?

File: src/app/components/LocationMap/MapwithMarker.tsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+import type { Map, LngLatLike } from 'mapbox-gl';
+import 'mapbox-gl/dist/mapbox-gl.css';
+import styled from 'styled-components';
+import MapboxGeocoder from '@mapbox/mapbox-gl-geocoder';
+import '@mapbox/mapbox-gl-geocoder/dist/mapbox-gl-geocoder.css';
+import type { GeoJsonType } from '../../utils/GeoJsonType';
+
+type MapWithMarkerProps = {
+  displayArea: string;
+  onChange: (event: any) => void;
```
Comment (PReichetanz) [inline src/app/components/LocationMap/MapwithMarker.tsx:12]: Is there a better type than any?

**Review by PReichetanz — COMMENTED** [review]
Looks good to me 😊

**Pull request merged**